velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -394,6 +394,46 @@ TEST_F(ArithmeticTest, atan2) {
   EXPECT_EQ(atan2(-1.0, -1.0), std::atan2(-1.0, -1.0));
 }
 
+TEST_F(ArithmeticTest, hex) {
+  const auto toHexWithVarbinary = [&](std::optional<std::string> value) {
+    return evaluateOnce<std::string>("hex(cast(c0 as varbinary))", value);
+  };
+  const auto toHexWithVarchar = [&](std::optional<std::string> value) {
+    return evaluateOnce<std::string>("hex(c0)", value);
+  };
+  const auto toHexWithBigint = [&](std::optional<int64_t> value) {
+    return evaluateOnce<std::string>("hex(c0)", value);
+  };
+
+  EXPECT_EQ(std::nullopt, toHexWithVarbinary(std::nullopt));
+  EXPECT_EQ("", toHexWithVarbinary(""));
+  EXPECT_EQ("537061726B2053514C", toHexWithVarbinary("Spark SQL"));
+  EXPECT_EQ("537061726B652153514C", toHexWithVarbinary("Spark\x65\x21SQL"));
+  EXPECT_EQ("616263", toHexWithVarbinary("abc"));
+  EXPECT_EQ("68656C6C6F20776F726C64", toHexWithVarbinary("hello world"));
+  EXPECT_EQ(
+      "48656C6C6F20576F726C642066726F6D2056656C6F7821",
+      toHexWithVarbinary("Hello World from Velox!"));
+
+  EXPECT_EQ(std::nullopt, toHexWithVarchar(std::nullopt));
+  EXPECT_EQ("", toHexWithVarchar(""));
+  EXPECT_EQ("537061726B2053514C", toHexWithVarchar("Spark SQL"));
+  EXPECT_EQ(
+      "537061726BE695B0E68DAE53514C", toHexWithVarchar("Spark\u6570\u636ESQL"));
+  EXPECT_EQ("616263", toHexWithVarchar("abc"));
+  EXPECT_EQ("68656C6C6F20776F726C64", toHexWithVarchar("hello world"));
+  EXPECT_EQ(
+      "48656C6C6F20576F726C642066726F6D2056656C6F7821",
+      toHexWithVarchar("Hello World from Velox!"));
```

Comment:
Does it make sense to combine `toHexWithVarbinary` and `toHexWithVarchar` as one lambda, and test both cases with one test call?

Reply:
hmm.. make no sense for me, I think current test cases are more readable.